### PR TITLE
Fix bug: boat attack expanding from all borders

### PR DESCRIPTION
**Pull request by Riprefro54**
Bug fix: Previously, when a boat landed on a target (either a bot or free land), the player would expand from every border as if it were a normal attack. Now, when a boat lands, it expands only from the landing point. This fix involved adding "borderTiles" to the AttackExecutor. "borderTiles" is a set that determines the origin of the attack. If it is set to null, the player’s border tiles are used as before. This change was implemented to make an infrastructure for adding/modding attack types in the future.

### CODE REVIEW

File: src/game/attack/AttackActionHandler.ts [hunk 1]
```diff
@@ -64,24 +64,26 @@ class AttackActionHandler {
 	 * Schedule an attack on an unclaimed territory.
 	 * @param player The player that is attacking.
 	 * @param troops The amount of troops that are attacking.
+	 * @param borderTiles The tiles from which the attack is executed. If it is null, it will be the player's border tiles by default.
 	 */
-	attackUnclaimed(player: Player, troops: number): void {
+	attackUnclaimed(player: Player, troops: number, borderTiles: Set<number> | null): void {
```
Comment (esk33) [inline src/game/attack/AttackActionHandler.ts]: `borderTiles` could have `null` as a default value, saving the need to retroactively pass `null` to `attackUnclaimed`

Comment (Riprefro54) [reply src/game/attack/AttackActionHandler.ts]: i thought the same but player also passed as null. idk why platz made it like this but i made the same

edit: oh wait this is a diffrent function yeah you are correct 

edit 2: wait it being diffrent function doesnt matter hmm idk actually

File: package-lock.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
     "": {
       "name": "warfront-client",
       "devDependencies": {
+        "@eslint/js": "^9.7.0",
```
Comment (platz1de) [inline package-lock.json]: Unrelated change (blame npm)

File: src/game/attack/AttackActionHandler.ts (hunk 1 above)
Comment (platz1de) [reply src/game/attack/AttackActionHandler.ts]: Adding a default value (of null) would be fitting here, since not specifying a border overwrite is the default case

File: src/game/attack/AttackExecutor.ts [hunk 3]
```diff
@@ -20,11 +21,13 @@ export class AttackExecutor {
...
 	 * @param target The player that is being attacked, or null if the target is unclaimed territory.
 	 * @param troops The amount of troops that are attacking.
+	 * @param borderTiles The tiles from which the attack is executed. If it is null, it will be the player's border tiles by default.
 	 */
-	constructor(player: Player, target: Player | null, troops: number) {
+	constructor(player: Player, target: Player | null, troops: number, borderTiles: Set<number> | null) {
 		this.player = player;
 		this.target = target;
 		this.troops = troops;
+		this.borderTiles = borderTiles;
```
Comment (platz1de) [inline src/game/attack/AttackExecutor.ts]: This can be passed directly to orderTiles here (without adding it as a property to this class)

**Comment by platz1de** [comment]
Thank you!
